### ISSUE TITLE
Payment surcharge error - Klarna

### ISSUE THREAD

**Issue by litsher** [mollie/magento2#352]
**Describe the bug**
When trying to pay with Karna Pay Later, a error occurs when the Payment Surcharge fixed amount is 1 euro.

**Used versions**
- *Magento Version2.3.1)*:
- *Open source*:
- *1.18.1*:

**To Reproduce**
1. Go to the site
2. try to buy a product
3. Choose "Klarna Pay Later"

**Expected behavior**
I can buy the product

**Actual behavior**
I cant buy the product

**Additional context**

main.CRITICAL: Exception message: Error executing API call (422: Unprocessable Entity): The amount of the order does not match the total amount from the order lines. Expected order amount to be €8.57 but got €9.57. Field: amount. Documentation: https://docs.mollie.com/guides/handling-errors
Trace: <pre>#1 Mollie\Payment\Model\Mollie->startTransaction() called at [app/code/Mollie/Payment/Controller/Checkout/Redirect.php:228]

**Comment by Frank-Magmodules** [comment]
Hi @litsher, thank you for the detailed error report. 

Just to be sure, this Payment Surcharge is added in the Mollie module, or is this added using a different module?
As we can't reproduce this when we use the Payment Surcharge function within the Mollie module.

**Comment by litsher** [comment]
Hey @Frank-Magmodules ,

This is added via the Mollie module, no other modules are involved in the payment part of the webshop.

**Comment by Frank-Magmodules** [comment]
HI @litsher , strange, I am not able to reproduce this on the [latest](https://github.com/mollie/magento2/releases) version of the extension. Can you please update the Mollie plugin to the latest version? In case if the issue is still there, please share your store-login details through our [contact form](https://www.magmodules.eu/support-form.html) so we can help you debug.

**Comment by litsher** [comment]
I have updated to the lastest version in a hope to fix this problem.
I will write to your contact information. If this is no bug in the most current version of the build this ticket can be closed.

Thanks for the quick response @Frank-Magmodules ! :)

**Comment by Frank-Magmodules** [comment]
HI @litsher , 

Do you have any update on this issue after our e-mail communication?
Please let me know if you need any further assistance so we can help you out with this issue.

**Comment by Frank-Magmodules** [comment]
Hi @litsher,

Due to inactivity, we are closing this issue now but please feel free to reopen the issue if this still occurs.

**Comment by Margero** [comment]
Hi,

Using the latest mollie version: 1.23.1
Magento Version 2.3.6

I have been getting this same error when using surcharge.
(view attachment)
![image error](https://user-images.githubusercontent.com/31563367/111280323-f6fe8580-863b-11eb-82ae-90d9633dc014.jpeg)

**Comment by litsher** [comment]
I must say that it seems to be an rounding error here, most likely tax related.

**Comment by Margero** [comment]
Error executing API call (422: Unprocessable Entity): Order line 29 is invalid. VAT amount is off. Expected VAT amount to be €0.54 (€3.14 × (21.00% / 121.00%)), got €0.55. Field: lines.29.vatAmount. Documentation: https://docs.mollie.com/guides/handling-errors

This is what I got from the logs. When doing the calculation myself I got 0,54495 so I think the rounding is being made starting from the 4th decimal by Mollie and from the 3th decimal by Magento?
